### PR TITLE
Smart constructors for -\/, \/-.

### PR DESCRIPTION
**Pull request by TomasMikula**

### CODE REVIEW

File: core/src/main/scala/scalaz/Either.scala [hunk 1]
```diff
@@ -370,10 +370,18 @@ final case class \/-[A, B](b: B) extends (A \/ B) {
 object \/ extends DisjunctionInstances {
 
   /** Construct a left disjunction value. */
+  def inl[A, B](a: A): A \/ B =
```
Comment (jbgi) [inline core/src/main/scala/scalaz/Either.scala]: I'd rather see `left` and `right` spelled out.

**Comment by jbgi** [comment]
alternatively, what about
```
object -\/ {
  def apply[A, B](a: A): A \/ B = new -\/(a)
}
```
This would preclude backports targeting scala 2.10 though.

**Comment by TomasMikula** [comment]
I didn't know you could do that. See #1645

**Comment by TomasMikula** [comment]
Closing in favor of #1645.

**Pull request closed without merging**